### ISSUE TITLE
Asset Contract Validation --  "status": "Contract file is invalid",

### ISSUE THREAD

**Issue by Ryanmtate** [coinspark/sparkbit#9]
Hi guys,

Using the CoinSpark API to issue an asset and send it to a SparkBit wallet, I am downloading the contract pdf using the result.contract_final object. The entire process takes ~5.04 seconds from API call to downloading the contract and storing locally.

When the asset is sent to the wallet, the balance shows that the contract is invalid:

```
    "status": "Contract file is invalid",
```

I have also tried to refresh the asset, but I still notice the contract shows invalid. The contract url for asset is available immediately upon download of the contract from the API callback. 

Not sure if I am encoding the download wrong, and thus the asset hash is different even because the slightest change to the pdf content. that's my current working hypothesis...

additional details:

```
{
    "asset_ref": "501353-5192-31783",
    "total": {
      "raw": 0,
      "qty": 0,
      "display": "0"
    },
    "spendable": {
      "raw": 0,
      "qty": 0,
      "display": "0"
    },
    "name": "vex.c",
    "name_short": "vex.c",
    "domain": "a.peermarket.exchange",
    "url": "http://a.peermarket.exchange/vex.c/",
    "issuer": "Venture Equity Exchange",
    "description": "cs vex.c",
    "units": "100 USD",
    "multiple": 10,
    "status": "Contract file is invalid",
    "valid": false,
    "checked_unixtime": 1436653560,
    "contract_url": "http://a.peermarket.exchange/vex.c/contract.pdf",
    "contract_file": "sparkbit.sparkwallet.csfiles/asset000018_contract.pdf",
    "genesis_txid": "277ca019269aaf0a5359ec911c58ca8a2b92267c070f097878fa94938d5feb3a",
    "added_unixtime": 1436644004,
    "issue_date": "11 July 2015, 7:46 pm",
    "issue_unixtime": 1436643999,
    "tracker_urls": [
      "http://assets-testnet.coinspark.org"
    ],
    "visible": true,
    "refreshing": false
  },
  {
    "asset_ref": "Awaiting new asset confirmation...",
    "total": {
      "raw": 0,
      "qty": 0,
      "display": "0"
    },
    "spendable": {
      "raw": 0,
      "qty": 0,
      "display": "0"
    },
    "name": "vex.e",
    "name_short": "vex.e",
    "domain": "a.peermarket.exchange",
    "url": "http://a.peermarket.exchange/vex.e/",
    "issuer": "Venture Equity Exchange",
    "description": "vex.e",
    "units": "100 USD",
    "multiple": 1,
    "status": "Contract file is invalid",
    "valid": false,
    "checked_unixtime": 1436654059,
    "contract_url": "http://a.peermarket.exchange/vex.e/contract.pdf",
    "contract_file": "sparkbit.sparkwallet.csfiles/asset000020_contract.pdf",
    "genesis_txid": "2f00e1772593dfc29f66e271280f1ca300a3fc9e2d27aa37b844e192e574f1e0",
    "added_unixtime": 1436653384,
    "issue_date": "11 July 2015, 10:23 pm",
    "issue_unixtime": 1436653380,
    "tracker_urls": [
      "http://assets-testnet.coinspark.org"
    ],
    "visible": true,
    "refreshing": false
  }

```

Thanks,

Ryan

**Comment by gidgreen** [comment]
I tried viewing the contract at http://a.peermarket.exchange/vex.e/contract.pdf in my browser, Adobe Reader, and Mac OS X's Preview app - and in all cases it does not show up any content.

However if I download your contract from the URL in the `contract_final` field of the API response (I'm not posting it here to protect your confidentiality) it displays fine.

So yes, it appears that something is going wrong when you are downloading the contract from that URL. I looked inside the two PDF files and it seems your code did a conversion of non-ASCII characters into `0xEF 0xBF 0xBD`. This suggests the PDF file has been run through some kind of character encoding converter, rather than being kept pristine in its binary form.
